homeassistant/components/philips_js/switch.py

```diff
@@ -70,3 +76,54 @@ async def async_turn_on(self, **kwargs: Any) -> None:
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the entity off."""
         await self.coordinator.api.setScreenState("Off")
+
+
+class PhilipsTVAmbilightHueSwitch(
+    CoordinatorEntity[PhilipsTVDataUpdateCoordinator], SwitchEntity
+):
+    """A Philips TV Ambi+Hue switch."""
+
+    def __init__(
+        self,
+        coordinator: PhilipsTVDataUpdateCoordinator,
+    ) -> None:
+        """Initialize entity."""
+
+        super().__init__(coordinator)
+
+        self._attr_name = f"{coordinator.system['name']} Ambilight+Hue"
+        self._attr_icon = "mdi:television-ambient-light"
+        self._attr_unique_id = f"{coordinator.unique_id}_ambi_hue"
+        self._attr_is_on = self.coordinator.api.huelamp_power == HUE_POWER_ON
+        self._attr_device_info = DeviceInfo(
+            identifiers={
+                (DOMAIN, coordinator.unique_id),
+            }
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return true if entity is available."""
+        if not super().available:
+            return False
+        if not self.coordinator.api.on:
+            return False
+        return self.coordinator.api.powerstate == "On"
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the entity on."""
+        await self.coordinator.api.setHueLampPower(HUE_POWER_ON)
+        self._attr_is_on = True
+        await self.coordinator.async_request_refresh()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the entity off."""
+        await self.coordinator.api.setHueLampPower(HUE_POWER_OFF)
+        self._attr_is_on = False
+        await self.coordinator.async_request_refresh()
```

Comment:
No need to request a refresh, just update from coordinator. Also this should call
self.async_write_ha_state() after it's data is updated.

Reply:
Thanks. This solved the issues with the switch toggling back to old state for a short duration.
Tested quite extensively by toggling on/off tens of times through HA, restarting HA multiple times. And toggling the Ambi+Hue with my remote and see if the state is reflected after the coordinator upgrade.